### PR TITLE
Add as struct

### PR DESCRIPTION
**Pull request by emceeaich**
Merges Errol's change for results as struct.

### CODE REVIEW

**Comment by emceeaich** [comment]
The two MacOS checks fail but the workflow file from upstream allows for them to fail:

```
          # db: A DB's brew package name in macOS case.
          #     Set a name "db: 'name@X.Y'" when using an old version.
          # MariaDB lastet version
          # Allow failure due to the following test failures that rarely happens.
          # https://github.com/brianmario/mysql2/issues/1194
          - {os: macos-latest, ruby: '2.6', db: mariadb, allow-failure: true}
          # MySQL latest version
          # Allow failure due to the issue #1194.
          - {os: macos-latest, ruby: '2.6', db: mysql, allow-failure: true}
```

File: ext/mysql2/result.c [hunk 1]
```diff
@@ -801,6 +831,7 @@ static VALUE rb_mysql_result_fetch_row(VALUE self, MYSQL_FIELD * fields, const r
                     offset = rb_funcall(cMysql2Client, intern_local_offset, 0);
                     val = rb_funcall(val, intern_new_offset, 1, offset);
                   } else { /* utc */
+                    VALUE defaults, opts, as_opt, block, (*fetch_row_func)(VALUE, MYSQL_FIELD *fields, const result_each_args *args);
```
Comment (danieldickison) [inline ext/mysql2/result.c]: What's this line about? Seems like maybe it's spurious?

**Review by danieldickison — COMMENTED** [review]
Can you comment on the one mysterious (to me) line?

File: ext/mysql2/result.c (hunk 1 above)
Comment (emceeaich) [reply ext/mysql2/result.c]: As removing that line has no effect on building or testing the gem, I believe it wound up a spurious inclusion from the merge of Errol's branch.

**Review by bcdanieldickison — APPROVED** [review]

**Pull request merged**